docs/tests.md

```diff
@@ -0,0 +1,415 @@
+# Tests
+
+## Testing Suite Architecture Overview
+
+This section provides an overview of the testing architecture for our Python
+library, which employs two types of tests: **regression tests** and
+**doctests**. Both types are powered by pytest, with doctests **(WIP)** being
+specifically based on the Python's doctest module.
+
+## Regression Tests: Understanding the Process
+
+At its core, our regression testing framework is designed to ensure that any
+changes made to the library don't inadvertently break existing functionality.
+Unlike doctests, which focus on individual components in isolation, regression
+tests take a broader approach by comparing the output of entire estimators
+against pre-computed expected results.
+
+This approach brings a significant advantage: **improved testing times.**
+Running each regression test, which often involves complex computations, can be
+very time-consuming. To optimize this process, we serialize expected outputs on
+predefined inputs. These serialized results are then used in subsequent test
+runs for efficient comparisons, saving valuable time during development.
+
+Here's a breakdown of how it works:
+
+```mermaid
+
+  flowchart LR;
+    gf("`Generators`")
+    inputs("`Inputs`");
+    preeo("`Precompute expected outputs`");
+    ser("`Serialize and store inputs/expected outputs`")
+    estimator("`Estimator`")
+    oc("`Test: Outputs comparison`")
+
+
+    gf --> preeo;
+    inputs --> preeo;
+    preeo --> ser;
+    ser -- input --> estimator;
+    estimator -- output --> oc;
+    ser -- expected output --> oc;
+
+```
```

Comment:
If possible, it would be good to add the dictionary name `REFERENCES_INPUTS` in the figure